storage/interface.go

```diff
@@ -157,6 +197,13 @@ type Appender interface {
 	Rollback() error
 }
 
+// ExemplarAppender provides an interface for adding samples to exemplar storage, which
+// within Prometheus is in-memory only.
+type ExemplarAppender interface {
+	// Add adds an exemplar to the for the given series labels.
+	AddExemplar(l labels.Labels, e exemplar.Exemplar) error
```

Comment:
Is this whole interface needed now?

Reply:
Hmmm this looks super confusing now - we have `Appended` with a `AppendExemplar` method, and  `ExemplarAppender` with an `AddExemplar` method.  I'm going to suggest we don't need both, and that one is potentially internal to TSDB?

Reply:
I've condensed/reduced the interfaces somewhat in the latest commit, let me know if this is in line with what you were thinking. I also fixed a bug I had introduced in the scrape code while rebasing/refactoring `AddExemplar/AddExemplarFast` --> `AppendExemplar`.

Reply:
Yeah LGTM